go/vt/vterrors/vterrors.go

```diff
@@ -244,6 +245,26 @@ func Wrapf(err error, format string, args ...any) error {
 	}
 }
 
+// Unwrap attempts to return the Cause of the given error, if it is indeed the result of a vterrors.Wrapf()
+// The function indicates whether the error was indeed wrapped. If the error was not wrapped, the function
+// returns the original error.
+func Unwrap(err error) (wasWrapped bool, unwrapped error) {
+	var w *wrapping
+	if errors.As(err, &w) {
+		return true, w.Cause()
+	}
+	return false, err
+}
+
+// UnwrapAll attempts to recursively unwrap the given error, and returns the most underlying cause
+func UnwrapAll(err error) error {
+	wasWrapped := true
+	for wasWrapped {
+		wasWrapped, err = Unwrap(err)
+	}
```

Comment:
I feel like we should probably add a limit here to prevent never getting out of here some odd reason.

Reply:
The logic is legit IMHO and the unit tests make good coverage. Please give this another thought. Would you want to see code that counts to `50` and then fails? Would that code look clearer or even more confusing? Please let me know and I'll do whatever you think.

Reply:
Yeah, what to do when this fails due to hitting the limit would be unknown. We have some protection from infinite recursion and would panic: https://github.com/golang/go/commit/757e0de89f80e89626cc8b7d6e670c0e5ea7f192

So I'm totally fine with it as-is.